dev/deps/spark-deps-hadoop-3-hive-2.3

```diff
@@ -154,6 +154,9 @@ json4s-scalap_2.13/4.0.7//json4s-scalap_2.13-4.0.7.jar
 jsr305/3.0.0//jsr305-3.0.0.jar
 jta/1.1//jta-1.1.jar
 jul-to-slf4j/2.0.16//jul-to-slf4j-2.0.16.jar
+kotlin-stdlib-jdk7/2.0.10//kotlin-stdlib-jdk7-2.0.10.jar
```

Comment:
Do we really need this? Currently, Spark only supports `JDK17` and `JDK21`
<img width="421" alt="image" src="https://github.com/user-attachments/assets/4f6a487f-2bfa-4fa1-83c9-472ef674c4e4">

Reply:
@panbingkun,

Good question! The kotlin-stdlib-jdk8 transitive dependency was pulled in by okhttp/4.12.0:

https://mvnrepository.com/artifact/com.squareup.okhttp3/okhttp/4.12.0

I am not a Kotlin expert, I have just done some research and based on the below finding kotlin-stdlib is enough:

Source: https://kotlinlang.org/docs/whatsnew18.html#updated-jvm-compilation-target

"_If you have explicitly declared kotlin-stdlib-jdk7 and kotlin-stdlib-jdk8 as dependencies in your build scripts, then you should replace them with kotlin-stdlib._"

I have updated this pull request, removed the kotlin-stdlib-jdk8 dependency and tested this change in a separate pull request: https://github.com/roczei/spark/pull/4, all unit tests have passed.

Other finding:  https://kotlinlang.org/docs/faq.html#which-versions-of-jvm-does-kotlin-target

"_Kotlin lets you choose the version of JVM for execution. By default, the Kotlin/JVM compiler produces Java 8 compatible bytecode. If you want to make use of optimizations available in newer versions of Java, you can explicitly specify the target Java version from 9 to 21._"